openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -1655,6 +1655,7 @@ static int cert_requested(SSL* ssl, X509** x509Out, EVP_PKEY** pkeyOut) {
             break;
     }
 #else
+    (void) ssl2_ctype;
```

Comment:
Option 2 won't work because we take the address of it, and it's allocated on the stack.

Option 1 works, but my version is overall less code.